WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ConfigurableBundleItemViewModel.swift

```diff
@@ -59,7 +60,20 @@ final class ConfigurableBundleItemViewModel: ObservableObject, Identifiable {
         isOptional = bundleItem.isOptional
         let isOptionalAndSelected = existingOrderItem != nil
         self.isOptionalAndSelected = isOptionalAndSelected
-        let quantity = existingOrderItem?.quantity ?? bundleItem.defaultQuantity
+        let quantity: Decimal = {
+            guard let orderItemQuantity = existingOrderItem?.quantity else {
+                return bundleItem.defaultQuantity
+            }
+            let parentOrderItemQuantity = max(existingParentOrderItem?.quantity ?? 1, 1)
+
+            // When the parent order item quantity is incremented without configuring the bundle, the quantity of the child items
+            // are not multiplied. Thus, we want to return the child item quantity when its quantity is smaller than the parent order
+            // item quantity.
+            guard orderItemQuantity > parentOrderItemQuantity else {
+                return orderItemQuantity
+            }
+            return orderItemQuantity * 1.0 / parentOrderItemQuantity
+        }()
```

Comment:
I'm a little confused by what we're doing here, and I think it's connected to a possible bug in wp-admin (which is the basis for this logic, if I understand correctly).

I _think_ what's expected is this:
1. If I increase the parent bundle quantity, the child bundle items will be automatically multiplied by the new quantity.
2. When I open the configuration screen, it will display the child bundle item quantities per bundle.

However, that isn't working in wp-admin or the app. In wp-admin, it seems to divide the child bundle item quantities by the parent bundle quantity (step 2) without having multiplied the child bundle items first (step 1). Here's an example:

https://github.com/woocommerce/woocommerce-ios/assets/8658164/2226fb78-4dae-42a0-9bef-fd37a470cd5a

Notice how after I increased the parent bundle quantity there, the child bundle item quantities didn't change, but the configure screen showed half the quantities. This seems like a bug on the web. However, when I set the child bundle items back to the expected quantities per bundle, they are multiplied as expected when the configuration is saved.

In the app, this new logic means that the child bundle items are never getting multiplied by the parent bundle quantity as expected. This causes the child bundle items to have their quantities calculated in unexpected ways on the configuration screen (some are divided by the parent bundle quantity while others aren't) and never to be correctly increased in the product list in the order.

Reply:
> Notice how after I increased the parent bundle quantity there, the child bundle item quantities didn't change, but the configure screen showed half the quantities. This seems like a bug on the web. However, when I set the child bundle items back to the expected quantities per bundle, they are multiplied as expected when the configuration is saved.

As you pointed out an unexpected and confusing behavior in core (also mentioned in Pe5pgL-3Vd-p2#quantity-of-bundle-child-order-items - I should have emphasized this more in the PR description), updating the quantity of the parent bundle item does **not** update the quantity of the child bundled items. The quantity of the child items is only updated when the order is updated with a bundle configuration at a later time.

Alejo suggested a workaround in pe5pgL-3Ze-p2#comment-3231, but it isn't a trivial change so I'll work on this as a separate issue https://github.com/woocommerce/woocommerce-ios/issues/11177. There's a chance the merchant updates the bundle quantity in core without visiting the bundle configuration form, which still results in the confusing quantity UX as you mentioned. I think we can consider this as an edge case, and the app is behaving the same as in core.

For testing this PR without the quantity update workaround https://github.com/woocommerce/woocommerce-ios/issues/11177, I'd suggest updating the bundle quantity in core and then visiting the configuration form to trigger the quantity update for its child items.